### PR TITLE
Add example code and cross-link to BufReader docs

### PR DESCRIPTION
**Pull request by mbrubeck**
r? @steveklabnik

### CODE REVIEW

File: src/libstd/io/buffered.rs [hunk 1]
```diff
@@ -27,6 +27,24 @@ use iter;
 /// For example, every call to `read` on `TcpStream` results in a system call.
 /// A `BufReader` performs large, infrequent reads on the underlying `Read`
 /// and maintains an in-memory buffer of the results.
+///
+/// # Examples
+///
+/// ```no_run
+/// use std::io::prelude::*;
+/// use std::fs::BufReader;
```
Comment (alexcrichton) [inline src/libstd/io/buffered.rs]: s/fs::BufReader/io::BufReader/ ?

**Comment by alexcrichton** [comment]
Nice! r=me with one minor typo

**Comment by steveklabnik** [comment]
:confetti_ball:

**Comment by mbrubeck** [comment]
Typo fixed.

**Pull request merged**